### PR TITLE
various bug fixes to support 23R2

### PR DESCRIPTION
**Pull request by maxcapodi78**

### CODE REVIEW

File: _unittest/conftest.py [hunk 1]
```diff
@@ -103,46 +103,63 @@
...
+            #     oDesktop = None
+            #     proj_list = []
+            # if oDesktop and not settings.non_graphical:
+            #     oDesktop.ClearMessages("", "", 3)
+            # if proj_list:
+            #     for proj in proj_list:
+            #         try:
+            #             oDesktop.CloseProject(proj)
+            #         except:
+            #             pass
```
Comment (Alberto-DM) [inline _unittest/conftest.py:145]: Can we remove the commented lines?

**Review by Alberto-DM — APPROVED** [review]
Looks good to me.
Recommended some changes.

File: pyaedt/application/Design.py [hunk 2]
```diff
@@ -401,7 +401,10 @@ def aedt_version_id(self):
 
     @property
     def _aedt_version(self):
-        return _retry_ntimes(10, self.odesktop.GetVersion)[0:6]
+        v = self.odesktop.GetVersion()
+        if v:
+            return v[0:6]
+        return
```
[the remark this answers is not in the document]
Comment (maxcapodi78) [reply pyaedt/application/Design.py]: done

File: .github/workflows/cpython_linux.yml [hunk 3]
```diff
@@ -58,10 +58,11 @@ jobs:
...
         timeout-minutes: 30
         run: |
-          export ANSYSEM_ROOT222=/apps/AnsysEM/v222/Linux64
-          export LD_LIBRARY_PATH=$ANSYSEM_ROOT222/common/mono/Linux64/lib64:$ANSYSEM_ROOT222/Delcross:$LD_LIBRARY_PATH
+          export ANS_NODEPCHECK=1
+          export ANSYSEM_ROOT231=/apps/AnsysEM/v231/Linux64
+          export LD_LIBRARY_PATH=$ANSYSEM_ROOT231/common/mono/Linux64/lib64:$ANSYSEM_ROOT231/Delcross:$LD_LIBRARY_PATH
           source .pyaedt_test_env/bin/activate
-          pytest -n 4 --durations=50  --dist loadfile -v _unittest
+          pytest --tx 6*popen --durations=50  --dist loadfile -v _unittest
```
[the remark this answers is not in the document]
Comment (maxcapodi78) [reply .github/workflows/cpython_linux.yml:65]: done

**Pull request merged**